WildMeOrg/codex-frontend
# Dex 823 password reset page

Implements the new password reset email endpoint in Forgot.jsp.
Creates a new password reset component with a new password reset endpoint.

QA notes:
- Blank text field for /forgot url should disable the button. Enter should work for button click.
- Currently, if you test on QA2, you'll have to hit up Jon for the URL he gets for the password reset email (I think he has it set up to catch outgoing email in test@wildme.org in the QA server). Navigate to the link he gives you; it will be in the format of: https://qa.dyn.wildme.io/auth/code/someLongCode.

- Make sure that you navigate to this URL in a browser session that doesn't think that you're logged in, or you'll get an error.

- When you navigate to this page, it will prompt you to enter a password and a confirmation password. Obfuscated input is forthcoming, as are details about valid password criteria. Both are beyond the scope of the current ticket. The submission button should be disabled both with empty text fields and when the password and confirm password don't match. You should be able to type enter to submit.

- You should get a success alert (or error alert); there should be a link in the text of the success. Note that the ticket spec called for automatic navigation to the login page, but when I implemented this, it was difficult to understand that it was because things had succeeded. I made the decision to guide the user from the success alert.

<img width="1552" alt="Screen Shot 2022-05-26 at 3 25 18 PM" src="https://user-images.githubusercontent.com/2775448/170591526-668e9c34-55e5-4fba-823b-96f04c769455.png">
<img width="1552" alt="Screen Shot 2022-05-26 at 3 25 27 PM" src="https://user-images.githubusercontent.com/2775448/170591543-28228d41-6b8f-4b01-bfd4-fcd2ca5e7321.png">
<img width="1552" alt="Screen Shot 2022-05-26 at 3 25 37 PM" src="https://user-images.githubusercontent.com/2775448/170591568-09891ff0-bb0b-4f32-9732-ad259b2c9cca.png">
<img width="1552" alt="Screen Shot 2022-05-26 at 3 26 27 PM" src="https://user-images.githubusercontent.com/2775448/170591594-5b851f9b-9f11-425a-8388-278c988a1364.png">
<img width="1552" alt="Screen Shot 2022-05-26 at 3 26 38 PM" src="https://user-images.githubusercontent.com/2775448/170591611-11b2d4ba-3990-41ba-8278-ccb5af58d322.png">
<img width="1552" alt="Screen Shot 2022-05-26 at 3 26 45 PM" src="https://user-images.githubusercontent.c
<img width="1552" alt="Screen Shot 2022-05-26 at 3 26 57 PM" src="https://user-images.githubusercontent.com/2775448/170591759-b9ccc41e-b52a-483b-b48e-c202dde1e121.png">
<img width="1552" alt="Screen Shot 2022-05-26 at 3 27 07 PM" src="https://user-images.githubusercontent.com/2775448/170591663-0eca7120-08f2-48e6-9b95-88d7d5e80dea.png">
